### PR TITLE
this will set maxDepth

### PR DESCRIPTION
**Pull request by falkben**
This isn't meant to be a complete fix, and still needs tweaking, but the following allows maxDepth param to be used.

A few things to note -- `returnDepthOfNode()` returns a value 1 less than the actual depth (_I'm pretty sure_)

The forest stats that print out w/ the calculated `maxDepth` don't account for this.  The average depth does account for this though, so you can get a `maxDepth: 1` w/ an `avg depth: 1.83` 🙄.  This should be fixed -- probably the best thing is for maxDepth to actually record the correct number in the first place.  Note that the logic in this PR would then need to be adjusted.

This logic for testing if it needs to be processed as a leaf node sounds like it should go into `shouldProcessNode` but I actually don't think that's correct (thus I removed your addition to that function).  However, it should probably go into _some_ function as this will get a bit messy if we leave it as is.

### CODE REVIEW

**Comment by falkben** [comment]
Also, note that I'm adding a small change to github's linguist to ignore the googletest files.

**Comment by falkben** [comment]
Now occasionally getting segfaults.  Have not yet been able to diagnose.  running bin/fp:

`9 1 4` works but `9 1 1` does not

**Comment by falkben** [comment]
call stack with a segfault:

```c++
fp::binStruct<double, int>::createRootChildNodes(fp::binStruct<double, int> * const this) (/usr/include/c++/8.2.1/bits/stl_vector.h:930)
fp::binStruct<double, int>::loadFirstNode(fp::binStruct<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/baseFunctions/fpBaseNode.h:80)
fp::binStruct<double, int>::createBin(int randSeed, int numTrees, fp::binStruct<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/forestTypes/binnedTree/binStruct.h:234)
fp::binnedBase<double, int>::growBins(fp::binnedBase<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/forestTypes/binnedTree/binnedBase.h:83)
fp::binnedBase<double, int>::growBins() [clone ._omp_fn.1](fp::binnedBase<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/forestTypes/binnedTree/binnedBase.h:116)
libgomp.so.1!GOMP_parallel(void (*)(void *) fn, void * data, unsigned int num_threads, unsigned int flags) (/build/gcc/src/gcc/libgomp/parallel.c:168)
fp::binnedBase<double, int>::growBins(fp::binnedBase<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/fpSingleton/fpInfo.h:154)
fp::binnedBase<double, int>::growForest(fp::binnedBase<double, int> * const this) (/home/ben/repos/RerF/packedForest/src/forestTypes/binnedTree/binnedBase.h:116)
fp::fpForest<double>::growForest(fp::fpForest<double> * const this) (/usr/include/c++/8.2.1/bits/unique_ptr.h:347)
main(int argc, char ** argv) (/home/ben/repos/RerF/packedForest/src/fp.cpp:112)
```

**Comment by MrAE** [comment]
It'd be awesome if all of the +1 fudge factors are taken care of.

File: packedForest/src/forestTypes/binnedTree/processingNode.h [hunk 1]
```diff
@@ -26,7 +26,7 @@ namespace fp{
 			protected:
 				int treeNum;
 				int parentNodeNumber;
-				int nodeNumber;
+				//int nodeNumber;
```
[the remark this answers is not in the document]
Comment (MrAE) [reply packedForest/src/forestTypes/binnedTree/processingNode.h]: I'd say so.

[the remark this answers is not in the document]
Comment (jbrowne6) [reply packedForest/src/forestTypes/binnedTree/processingNode.h]: agreed.

**Comment by jbrowne6** [comment]
I wouldn't say that the +1's are fudge factors.  They are required to say that a child's depth is its parents depth +1.  Without a plus one then everyone's depth will always be 0.

**Pull request merged**